src/Commands/AliasesCommand.php

```diff
@@ -2,38 +2,266 @@
+     *
+     * @return string
+     */
+    protected function shortenHomePath($message)
+    {
+        return str_replace($this->getConfig()->get('user_home'), '~', $message);
+    }
+
+    /**
+     * Fetch those sites indicated by the commandline options.
+     *
+     * @param array $options
+     *   Full set of commanline options, some of which may affect selected set
+     *   of sites returned.
+     *
+     * @return array
+     *   Associative array of site id => site name
      */
-    public function aliases($options = ['print' => false, 'location' => null,])
+    protected function getSites($options)
     {
-        $aliases = $this->session()->getUser()->getAliases();
-        if (isset($options['print']) && $options['print']) {
-            return $aliases;
+        if (!empty($options['only'])) {
+            return $this->getSpecifiedSites(explode(',', $options['only']));
         }
-        if (is_null($location = $options['location'])) {
-            $location = '~/.drush/pantheon.aliases.drushrc.php';
+        if (!$options['all']) {
+            return $this->getSitesWithDirectMembership();
+        }
+        return $this->getAllSites($options);
+    }
+
+    /**
+     * Fetch the sites listed on the command line.
+     *
+     * @param array $siteList
+     *   List of site names
+     *
+     * @return array
+     *   Associative array of site id => site name
+     */
+    protected function getSpecifiedSites($siteList)
+    {
+        $result = [];
+        foreach ($siteList as $siteName) {
+            $site = $this->sites()->get($siteName);
+            $result[$site->id] = $siteName;
+        }
+        return $result;
+    }
+
+    /**
+     * Look up all available sites, as filtered by --org and --team
+     *
+     * @param array $options
+     *   Full set of commanline options, some of which may affect selected set
+     *   of sites returned.
+     *
+     * @return array
+     *   Associative array of site id => site name
+     */
+    protected function getAllSites($options)
+    {
+        $this->sites()->fetch(
+            [
+                'org_id' => null,
+                'team_only' => false,
+            ]
+        );
+        return $this->getSiteNames($this->sites->ids());
+    }
+
+    /**
+     * Look up those sites that the user has a direct membership in
+     *
+     * @return array
+     *   Associative array of site id => site name
+     */
+    protected function getSitesWithDirectMembership()
+    {
+        $this->sites()->fetch(
+            [
+                'org_id' => null,
+                'team_only' => true,
+            ]
+        );
+        return $this->getSiteNames($this->sites->ids());
+    }
+
+    /**
+     * Given a set of site ids, return an id=>name mapping.
+     *
+     * @param array $site_ids
+     *   List of site ids
+     *
+     * @return array
+     *   Associative array of site id => site name
+     */
+    protected function getSiteNames($site_ids)
```

Comment:
I think could be done more efficiently via
`array_map(function ($siteInfo) {return $siteInfo['name'];}, $this->sites->serialize());`